projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1952,7 +1952,11 @@ export class IgxTimePickerComponent implements
 
             const oldVal = new Date(this.value);
 
-            this.displayValue = '';
+            this.displayValue = this.parseMask(false);
+            requestAnimationFrame(() => {
+                this._setCursorPosition(0);
+            });
+            // TODO: refactoring - this.value should be null #8135
             this.value.setHours(0, 0);
```

Comment:
Two issues here, probably related with #8135

- if I start with time picker in drop down `mode` with no value, e.g. `<igx-time-picker [mode]="'DropDown'"></igx-time-picker>` this on first place has `clear` button. On second place if I click `clear` button I end up with error `Cannot read property 'setHours' of undefined`. Same behavior on master!
- if I have value, press back space until value is completely deleted. Then I enter some value and press the `clear` button value is not cleared. It is only moves the cursor to the start of input. This does not happen on master, where the value is cleared.

Again, not sure if this is related to above mentioned issue, but if this is, IMO we should first fix the other issue.